src/sidebar/components/hooks/use-user-filter-options.js

```diff
@@ -0,0 +1,79 @@
+import { useMemo } from 'preact/hooks';
+
+import { useStoreProxy } from '../../store/use-store';
+import { username } from '../../util/account-id';
+
+/** @typedef {import('../../store/modules/filters').FilterOption} FilterOption */
+
+/**
+ * Generate a list of users for filtering annotations; update when set of
+ * annotations or filter state changes meaningfully.
+ *
+ * @return {FilterOption[]}
+ */
+export default function useUserFilterOptions() {
```

Comment:
I'd like some feedback on whether a hook is the best way to encapsulate this logic. We need to think about: How would this adapt/be replaced if we relied on a (backend) service to provide the options for this filter facet, e.g.?

Reply:
I think a hook is a natural choice here. Custom hooks are good for encapsulating the logic for fetching data from some data source and triggering a re-render of a UI component when that data source changes. The main decision point is what part of the logic goes in the hook and what goes in the calling component.